src/main/scala/beam/sim/vehiclesharing/RepositionManagerListener.scala

```diff
@@ -0,0 +1,74 @@
+package beam.sim.vehiclesharing
+
+import beam.agentsim.agents.vehicles.BeamVehicle
+import beam.agentsim.infrastructure.TAZTreeMap.TAZ
+import org.matsim.api.core.v01.Id
+import org.matsim.core.controler.events.IterationEndsEvent
+import org.matsim.core.controler.listener.{ControlerListener, IterationEndsListener}
+import org.matsim.core.utils.io.IOUtils
+
+import scala.collection.mutable
+
+class RepositionManagerListener extends ControlerListener with IterationEndsListener {
+
+  override def notifyIterationEnds(event: IterationEndsEvent): Unit = {
+    if (eventsList.nonEmpty)
+      RepositionManagerListener.writeRepositionEvents(event, eventsList)
+  }
+
+  val eventsList = mutable.ListBuffer.empty[(Int, Id[TAZ], Id[VehicleManager], Id[BeamVehicle], String, String)]
```

Comment:
How many events will be here? Could it cause an issue (memory) on production run with a lot of vehicles? If it is stored only for the purpose of writing it in the end of iteration, it can be written during the simulation instead of accumulating all events

Reply:
so this is the big change I did since yesterday. I got rid of the whole listener thing and rather used the skimmer with more or less same purpose